test/parallel/test-stream-writable-end-cb-error.js

```diff
@@ -46,3 +46,38 @@ const stream = require('stream');
     writable.emit('error', new Error('kaboom'));
   }));
 }
+
+{
+  const w = new stream.Writable({
+    write(chunk, encoding, callback) {
+      setImmediate(callback);
+    },
+    finish(callback) {
+      setImmediate(callback);
+    }
+  });
+  w.end('testing ended state', common.mustCall((err) => {
+    // This errors since .destroy(err), which is invoked by errors
+    // in same tick below, will error all pending callbacks.
+    // Does this make sense? Not sure.
```

Comment:
In my opinion it makes sense for pending callbacks but not for the callback of the in progress write.